Description: Add get_link_layer support to rdmavt
 Fixes a kernel crash in 3.16.51-3+deb8u1 due to NULL function pointer
 for the get_link_layer verb.
Author: Brian T. Smith <bsmith@systemfabricworks.com>
Forwarded: not-needed
Last-Update: <2018-09-17>
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/rdmavt/vt.c
+++ b/rdmavt/vt.c
@@ -342,6 +342,12 @@
 }
 #endif /* CONFIG_RVT_GET_PORT_IMMUTABLE */
 
+static enum rdma_link_layer rvt_get_link_layer(struct ib_device *device,
+											   u8 port_num)
+{
+	return IB_LINK_LAYER_INFINIBAND;
+}
+
 enum {
 	MISC,
 	QUERY_DEVICE,
@@ -387,6 +393,7 @@
 	RESIZE_CQ,
 	ALLOC_PD,
 	DEALLOC_PD,
+	GET_LINK_LAYER,
 	_VERB_IDX_MAX /* Must always be last! */
 };
 
@@ -714,6 +721,12 @@
 				      rvt_dealloc_pd);
 		break;
 
+	case GET_LINK_LAYER:
+		check_driver_override(rdi, offsetof(struct ib_device,
+											get_link_layer),
+							  rvt_get_link_layer);
+		break;
+
 	default:
 		return -EINVAL;
 	}
